blockchain/blockchain.go

```diff
@@ -450,54 +450,90 @@ func (bc *BlockChain) loadLastState() error {
 	return nil
 }
 
-// SetHead rewinds the local chain to a new head. In the case of headers, everything
-// above the new head will be deleted and the new one set. In the case of blocks
-// though, the head may be further rewound if block bodies are missing (non-archive
-// nodes after a fast sync).
 func (bc *BlockChain) SetHead(head uint64) error {
```

Comment:
There is no **specific** reference version of this function.
Because necessary features are spread over several versions, and function format is different with Ethereum.
But mostly from these following PR --> https://github.com/ethereum/go-ethereum/commit/8cbdc8638fd28693f84d7bdbbdd587e8c57f6383

+ I'll add some comments about setHead. Don't remember why I just deleted. Thanks for reminding.